packages/wonder-blocks-core/util/unique-id-factory.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import {IIdentifierFactory} from "../util/types.js";
+
+/**
+ * This is NOT for direct use. Instead, see the UniqueIDProvider component.
+ *
+ * Implements IIdentifierFactory to provide unique identifiers.
+ */
+export default class UniqueIDFactory implements IIdentifierFactory {
```

Comment:
The naming of this is a little confusing.  I would expect a factory of unique ids to be able to return multiple ids each of which are unique.  If I run the following code:
```
const uniqueIdFactory = new UniqueIDFactory("foo");
const id1 = uniqueIdFactory.id("bar");
const id2 = uniqueIdFactory.id("bar");
```
I would expect `id1 !== id2` to be true, but that isn't actually the case.  If we want to use the factory pattern there should be a singleton that call `UniqueIDFactory` that returns instances of this class and this class should have a different name.

Reply:
That's not right at all.

The point here is that the factory produces unique identifiers for a given instance. If we ask for `id("bar")` it will give us a unique ID for "bar". If we ask for `id("foo")` it will give us a unique ID for id "foo". Calls to ask for that ID will then always return the same identifier, allowing us to just have `id("foo")` in our code and not care about what it really means. 

That's the pattern we established for webapp and for ADR 60 and the prior ADR 58.

Reply:
Thanks for pointing back to ADR 58.  Sorry I didn't mean to rehash previous decisions.  I do find the naming of things a bit confusing, but the tests and documentation and help.  Thanks for calling out that calling `.get("test")` will always return the same value.

Reply:
Naming things is hard. I'm not particularly hooked on the names, so we could change them if we think of something better.